### ISSUE TITLE
Unable to see/access fourth hdf drive in Workbench (amigavision setup)

### ISSUE THREAD

**Issue by lunchpacked130** [Mazamars312/Analogue-Amiga#46]
I'm running the amigavision/megaags hdf files as the first two drives, but i have 2 additional hdf drives i aøso want to access in workbench. (I have not created these drives, so unsure what filesystem it uses or how it was created)

I renamed the two additional hdf files to "HDD-3.hdf" and "HDD-4.hdf" respectively, I even copy pasted the names from data.json to make sure it works without any naming issues, also made sure the json settings for the drives are identical to the first to drives (except filenames of course)

After starting the amiga core on the Analogue Pocket and exiting megaags to workbench however, I would only have HDD-4 show up.
When checking on a computer with winuae, noticed the drive icon was a standard icon, so i assumed maybe its a different filesystem or similar.

To try to fix this, i made a copy of the known working HDD-4.hdf file, loaded it up in a winuae workbench environment together with HDD-3.hdf,
Manually deleted all files from HDD-4, and copied everything from HDD-3 over to HDD-4, and renamed drive name in workbench and hdf filename in windows explorer to match the old HDD-3
Replaced the old HDD-3.hdf file on the AP SD card with the new one.

Now, when booting on the AP, only HDD-3 shows up in workbench, and HDD-4 is suddenly missing.

I have checked HDToolbox, and it shows 4 devices, two identical with unknown hdf names and one with the original hdf name (not the renamed HDD-x name), but I'm not familiar enough with amiga HDD/hdf configs other than in my experience, any changes made there results in having to format the drive again.

Am I doing something wrong? Can there be a conflict with the partitions used by amigavision/megaags?

Summary of issue:
When trying to load HDD-3.hdf and HDD-4.hdf, only one of those two drives show up in workbench (IDEFix97 installed)

Expected behaviour:
All drives should show up in workbench.

How to recreate:
(Assuming amigavision/megaags with IDEFix97 installed and running in Amiga core on Analogue Pocket, not using a dock (as i don'thave one))
Have HDD-3.HDF and HDD-4.hdf files in "common" folder.
From MegaAGS menu, press [Select] on the AP to bring up onscreen keyboard and select "Esc" to exit to workbench.
In workbench there should be 2 additional drives available, but i only get one additional drive to show up.

For clarification:
● the 4 drives i have set up to load is: "MegaAGS.hdf", "MegaAGS-Saves.hdf", "HDD-3.hdf" and "HDD-4.hdf".
● I have the newest version of the Amiga core installed
● I have tried reinstalling the Amiga core.
● I'm using the 9gb AmigaVision/MegaAGS hdf file.
● I made sure all settings and names are correct and matching as described in the readme and data.json
● I have the newest IDEFix97 installed.
● I have tried reinstalling IDEFix97.
● I have checked to see if the drive icon is hidden, either by system, behind another icon or outside the viewable desktop
● I have *not* made any changes in HDToolbox
● I tried getting Atapimagic to work as it was supposed to have better compatability then idefix97, but was unable to make it detect any extra drives. 
(Does anyone have a setup guide / startup-sequence file that works with the amigavision/megaags setup?)

**Comment by limi** [comment]
I doubt this was tested on the Pocket core, as it requires quite a bit of Amiga-specific knowledge (IDEFix97 etc), as you already know. We ended up not including anything like that with AmigaVision since it was causing some other problems, but it does work on the MiSTer.

**Comment by lunchpacked130** [comment]
I never managed to get the core to work with 4 harddrives, so as a workaround, i made a single hdf drive a smidhe over twice as large as the original hdf files and partitioning it wirh two drives instead

Functionally, the result is exactly the same
I also made the new drive in pfs3 filesystem instead of FFS which the old drives used  and everything works beautifully.

With IDEFIX97 installed, mounting 3 drives always works. But the 4th drive seems to be the on that causes issues. (Drives mentioned here by connections, not specific drives)

I also noticed some online mentioning that idefix97 itself could have some issues mounting drives correctly, and that atapi magic is one that fixes /does the same thing only working what idefix97 has issues with.

Anyway, it seems this fault can just as well be idefix97 as it could be the core being the one having issues,  so i will close this thread as "not planned" as the issue was not resolved (and may never be) and workaround works perfectly.

Maybe the readme should be updated with there might be issues getting 4 hdf files to work, and if possible, using a large hdf file and partition it instead.
Though this might involve copying files inside amiga workbench, which can take some time depending on the size, but it was much faster copying around 6gb in dopus than downloading even all the extra asset roms for CPS2 core 😅 and you only have to do it once
